### PR TITLE
drivers/tone_alarm and tune_control small improvements/cleanup

### PR DESCRIPTION
**Pull request by dagar**
- drivers/tone_alarm: move to ModuleBase and purge CDev (/dev/tone_alarm0)
 - drivers/tone_alarm: only run on tune_control publication (or scheduled note) rather than continuously
 - drivers/tone_alarm: use HRT to schedule tone stop (prevents potential disruption)
 - msg/tune_control: add tune_id numbering
 - systemcmds/tune_control: add "error" special case tune_id
 - move all tune_control publication to new uORB::PublicationQueued<>
 - start tone_alarm immediately after board defaults are loaded to fix potential startup issues
 - for SITL (or other boards with no TONE output) print common messages (startup, error, etc)

### CODE REVIEW

File: src/drivers/tone_alarm/CMakeLists.txt [hunk 1]
```diff
@@ -34,8 +34,11 @@
 px4_add_module(
 	MODULE drivers__tone_alarm
 	MAIN tone_alarm
+	COMPILE_FLAGS
+		#-DDEBUG_BUILD
 	SRCS
 		ToneAlarm.cpp
+		ToneAlarm.h
```
Comment (mcsauder) [inline src/drivers/tone_alarm/CMakeLists.txt:41]: @dagar , I can build successfully without this line and bench tests show no difference.

**Comment by mcsauder** [comment]
Hi @dagar , I've checked through the list by enum value and alias and everything is working as expected.  I haven't uncovered anything that behaves improperly, and the functionality improves on the readability of the scripts and the functionality of code in general.  This looks like a great PR to get in from my testing.  Nice work!

**Comment by mcsauder** [comment]
One small difference as shown:
this PR with successful tune publishing/playing of the following:
image

Current master branch publishes info, which I don't think is a loss:
image

**Review by mcsauder — APPROVED** [review]
All of the bench testing I have performed functions as expected.  I think this is good to go.  Nice work.

-Mark

File: src/drivers/tone_alarm/CMakeLists.txt (hunk 1 above)
Comment (dagar) [reply src/drivers/tone_alarm/CMakeLists.txt:41]: It only exists to help IDEs with cmake integration.

**Pull request merged**